src/supplemental_constructors.jl

```diff
@@ -1,91 +1,123 @@
 
-"""Constructs Deterministic from a InfrastructureSystemsType, label, and TimeArray."""
-function Deterministic(component::InfrastructureSystemsType, label::String, data::TimeSeries.TimeArray)
-    resolution = get_resolution(data)
-    initial_time = TimeSeries.timestamp(data)[1]
-    Deterministic(component, label, Dates.Minute(resolution), initial_time, data)
+function make_public_forecast(forecast::ForecastInternal, d::TimeSeriesData)
+    error("$(typeof(forecast)) must implement make_public_forecast")
+end
+
+function make_internal_forecast(forecast::Forecast)
+    error("$(typeof(forecast)) must implement make_internal_forecast")
 end
 
-"""Constructs Deterministic after constructing a TimeArray from initial_time and time_steps.
 """
-function Deterministic(component::InfrastructureSystemsType,
-                       label::String,
+    Deterministic(label::String,
+                  resolution::Dates.Period,
+                  initial_time::Dates.DateTime,
+                  time_steps::Int)
+
+Constructs Deterministic after constructing a TimeArray from initial_time and time_steps.
+"""
+function Deterministic(label::String,
                        resolution::Dates.Period,
                        initial_time::Dates.DateTime,
                        time_steps::Int)
     data = TimeSeries.TimeArray(
         initial_time : Dates.Hour(1) : initial_time + resolution * (time_steps-1),
         ones(time_steps)
     )
-    return Deterministic(component, label, Dates.Minute(resolution), initial_time, data)
+    return Deterministic(label, data)
 end
 
-function Deterministic(component::InfrastructureSystemsType,
-                       label::AbstractString,
-                       resolution::Dates.Period,
-                       initial_time::Dates.DateTime,
-                       data::TimeSeries.TimeArray,
-                      )
-    start_index = 1
-    horizon = length(data)
-    return Deterministic(component, label, resolution, initial_time, data, start_index,
-                         horizon, InfrastructureSystemsInternal())
+function make_public_forecast(forecast::DeterministicInternal, data::TimeSeries.TimeArray)
+    return Deterministic(get_label(forecast), data)
 end
 
-"""Constructs Probabilistic after constructing a TimeArray from initial_time and time_steps.
+function make_internal_forecast(forecast::Deterministic, ts_data::TimeSeriesData)
```

Comment:
if internal forecasts are not meant to be manipulated by users, shouldn't this be internal.

Reply:
Not sure what you mean.  The function is not exported.  If you are referring to lack of a leading underscore, my personal policy is to lead with an underscore if a function is only used within a file (also a Python convention).  I may not have always strictly followed that, though.

Reply:
I have tried to implement that all non-exported functions carry the _ in the beginning for clarity.

Reply:
We should have a rule for team consistency. A leading underscore is common for many people if the function is in the same file.  I find it a bit distracting if it's jumping across files.  Judging whether it's public requires an extra search, but that's ok with me.  That is very similar to other languages.  Python is the only language I know of where the underscore is the telling factor.